src/components/published-modal.cjsx

```diff
@@ -0,0 +1,46 @@
+React = require 'react'
+BS = require 'react-bootstrap'
+_ = require 'underscore'
+
+{ExerciseStore} = require '../stores/exercise'
+
+
+PublishedModal = React.createClass
+  getInitialState: -> {}
+
+  componentWillMount: ->
+    ExerciseStore.on('published', @show)
+
+  componentWillUnmount: ->
+    ExerciseStore.off('published', @show)
+
+  show: ->
+    @setState(isShowing: true)
+
+  onHide: ->
+    @setState(isShowing: false)
```

Comment:
yes, the modal is shown whenever `ExerciseStore` emits `published` (line 12).  it doesn't really care what exercise emits the event though,  I guess technically it could be a different exercise than the one that is set on `props.exerciseId`  I'll have it check the id in emit and only set `isShowing` to `true` if they match.